salt/modules/yumpkg.py

```diff
@@ -2582,7 +2582,7 @@ def mod_repo(repo, basedir=None, **kwargs):
 
     # Build a list of keys to be deleted
     todelete = []
-    for key in repo_opts:
+    for key in list(repo_opts):
```

Comment:
I think some inline documentation here about why this is required would help us avoid confusion in the future.